L/libjulia/common.jl

```diff
@@ -0,0 +1,106 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg.Types
+
+# Collection of sources required to build GMP
+function configure(version)
+    name = "libjulia"
+
+    checksums = Dict(
+        v"1.3.1" => "3d9037d281fb41ad67b443f42d8a8e400b016068d142d6fafce1952253ae93db",
+        v"1.4.2" => "76a94e06e68fb99822e0876a37c2ed3873e9061e895ab826fd8c9fc7e2f52795",
+        v"1.5.1" => "1f138205772eb1e565f1d7ccd6f237be8a4d18713a3466e3b8d3a6aad6483fd9",
+    )
+    sources = [
+        ArchiveSource("https://github.com/JuliaLang/julia/releases/download/v$(version)/julia-$(version).tar.gz", checksums[version]),
+    ]
+
+    # Bash recipe for building across all platforms
+    script = raw"""
+    cd $WORKSPACE/srcdir/julia*
+
+    FLAGS=(
+        USE_SYSTEM_LLVM=1
+        USE_SYSTEM_LIBUNWIND=1
+        USE_SYSTEM_PCRE=1
+        USE_SYSTEM_OPENLIBM=1
+        USE_SYSTEM_DSFMT=1
+        USE_SYSTEM_BLAS=1
+        LIBBLASNAME=libopenblas
+        USE_SYSTEM_LAPACK=1
+        LIBLAPACKNAME=libopenblas
+        #USE_SYSTEM_GMP=1
+        USE_SYSTEM_MPFR=1
+        USE_SYSTEM_SUITESPARSE=1
+        USE_SYSTEM_LIBUV=1
+        USE_SYSTEM_UTF8PROC=1
+        USE_SYSTEM_MBEDTLS=1
+        USE_SYSTEM_LIBSSH2=1
+        USE_SYSTEM_CURL=1
+        USE_SYSTEM_LIBGIT2=1
+        USE_SYSTEM_PATCHELF=1
+        USE_SYSTEM_ZLIB=1
+        USE_SYSTEM_P7ZIP=1
+
+        NO_GIT=1
+        prefix="${prefix}"
+        -j${nproc}
+    )
+
+    # compile libjulia but don't try to build a sysimage
+    make "${FLAGS[@]}" julia-ui-release
+    # 'manually' install libraries and headers
+    cp usr/lib/libjulia* ${libdir}/
```

Comment:
@staticfloat Thanks for working on this, too! However, I am concerned about several potential issues with the above lines:
- we need the headers, too (I tried to take care of that now)
- it omits a bunch of things `make install` does, e.g.:
   - for FreeBSD, it omits the calls to `fixup-rpath.sh`; this may or may not be an issue, as I don't know much about FreeBSD
   - for Windows, the DLL ought to go into `bin/`, I think (again I am not sure how much, if at all, this matters, but...)
   - for macOS, it doesn't copy the `*.dSYM` files (but I think that's fine)

Perhaps the latter are all OK, I just wanted to mention them all explicitly to very.